scheduler/scheduler.go

```diff
@@ -220,10 +285,101 @@ func spawnHosts(ctx context.Context, newHostsNeeded map[string]int) (map[string]
 		"span":      time.Since(startTime).String(),
 		"duration":  time.Since(startTime),
 	})
-
 	return hostsSpawnedPerDistro, nil
 }
 
+// IsMaxHostsRunning returns true if the max number of hosts are already running
+func IsMaxHostsRunning(d distro.Distro, allDistroHosts []host.Host) bool {
```

Comment:
You shouldn't need this here - the host allocator should handle not asking for more than max hosts (if we end up with slightly more than max due to some race that's fine)